tests/vmi_multiqueue_test.go

```diff
@@ -123,5 +123,35 @@ var _ = Describe("[Serial]MultiQueue", func() {
 			}
 		})
 
+		It("should be able to create a multi-queue VMI when requesting a single vCPU", func() {
+			vmi := tests.NewRandomFedoraVMIWitGuestAgent()
+			multiQueue := true
+
+			// by default, we request a single vCPU
+			vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue = &multiQueue
+			vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
+
+			By("Creating and starting the VMI")
+			vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 360)
+
+			By("Checking if we can login")
+			expecter, err := tests.LoggedInFedoraExpecter(vmi)
+			defer expecter.Close()
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Fetching Domain XML from running pod")
+			domain, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+			Expect(err).ToNot(HaveOccurred())
+			domSpec := &api.DomainSpec{}
+			Expect(xml.Unmarshal([]byte(domain), domSpec)).To(Succeed())
+
+			for _, iface := range domSpec.Devices.Interfaces {
+				Expect(iface.Target.Device).To(Equal("tap0"), "the target name should be tap0")
+				Expect(iface.Target.Managed).To(Equal("no"), "we should instruct libvirt not to configure the tap device")
+			}
```

Comment:
Don't you need to check about the queue somehow?

Reply:
No need; if the VMI exists, it is because libvirt was able to consume the pre-created tap device in single-queue mode.

Actually, just checking the VMI was created without errors would be enough, but I think that section makes it a bit more explicit, and in line with the remaining tests on this file.

Reply:
This is strange. You assume the queue is indeed consume and translated correctly to the domain config and that the domain interprets it correctly and does what it needs. I am usually trying to avoid assumptions as they bite later on.

Reply:
> You assume the queue is indeed consume and translated correctly to the domain config and that the domain interprets it correctly and does what it needs.

If the tap device fails to be consumed, 'creating' the VMI will fail, as it throws a warning, which will cause `WaitForSuccessfulVMIStartWithTimeout` to fail.

Reply:
I am pointing out that I think it will be safer to check that what you expect here (single queue configuration) is in place and not something else. It is for sure more interesting than if the device is managed or not.
But if you are sure no corner cases may occur and it is safe as is, then I will not insist.